astropy/io/fits/hdu/table.py

```diff
@@ -999,14 +1009,27 @@ def dump(self, datafile=None, cdfile=None, hfile=None, clobber=False):
         plain text (ASCII) files.
         """
 
+        if isinstance(clobber, bool) and isinstance(overwrite, bool):
```

Comment:
This seems identical to the code used in `report`. Maybe you could factor the code this code into a seperate function?